bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableOptionsFactory.java

```diff
@@ -144,8 +148,28 @@
       "google.bigtable.grpc.channel.timeout.ms";
   public static final long BIGTABLE_CHANNEL_TIMEOUT_MS_DEFAULT = 30 * 60 * 1000;
 
-  public static BigtableOptions fromConfiguration(Configuration configuration) throws IOException {
-    BigtableOptions.Builder bigtableOptionsBuilder = new BigtableOptions.Builder();
+  public static BigtableOptions fromConfiguration(final Configuration configuration)
+      throws IOException {
+
+    final BigtableOptions.Builder bigtableOptionsBuilder = new BigtableOptions.Builder();
+
+    final ScheduledExecutorService retryExecutor =
+        Executors.newScheduledThreadPool(
+            RETRY_THREAD_COUNT,
+            new ThreadFactoryBuilder()
+                .setDaemon(true)
+                .setNameFormat(RETRY_THREADPOOL_NAME + "-%d")
+                .build());
+
+    Future<Void> channelOptionFuture = retryExecutor.submit(new Callable<Void>() {
```

Comment:
Maybe it will be clearer if this is a Future<ChannelOptions> and once it is created bigtableOptionsBuilder.setChannelOptions i called? Afterall, I assume the builder is not threadsafe per se (although I dont think this would be a problem here).